packages/api/cms-api/src/redirects/import-redirects.console.ts

```diff
@@ -0,0 +1,217 @@
+                    this.repository.assign(existingRedirect, {
+                        target: this.linkBlock
+                            .blockInputFactory({
+                                attachedBlocks: [
+                                    {
+                                        type: "external",
+                                        props: {
+                                            targetUrl: row["target"],
+                                            openInNewWindow: true,
+                                        },
+                                    },
+                                ],
+                                activeType: "external",
+                            })
+                            .transformToBlockData(),
+                    });
+
+                    success++;
+
+                    await this.repository.flush();
+                } else {
+                    const redirect = this.repository.create({
+                        id: uuid(),
+                        sourceType: RedirectSourceTypeValues.path,
+                        source: row["source"],
+                        target: this.linkBlock
+                            .blockInputFactory({
+                                attachedBlocks: [
+                                    {
+                                        type: "external",
+                                        props: {
+                                            targetUrl: row["target"],
+                                            openInNewWindow: true,
+                                        },
+                                    },
+                                ],
+                                activeType: "external",
+                            })
+                            .transformToBlockData(),
+                        active: true,
+                        generationType: RedirectGenerationType.manual,
+                        createdAt: new Date(),
+                        updatedAt: new Date(),
+                        comment,
+                        scope: row["scope"],
+                    });
+
+                    success++;
+                    this.repository.persist(redirect);
+                }
+            } else {
+                console.log(`Error for Redirect ${row["source"]}`);
+                error++;
+            }
+        }
+
+        await this.repository.flush();
+        console.log(`\nSuccess: ${success}`);
+        console.log(`Error: ${error}`);
+    }
+
+    readRedirectsCsv = async (
+        filePath: string,
+    ): Promise<
+        Array<{
+            target: string;
+            target_type: string;
+            source: string;
+            comment: string;
+            scope: Scope;
+        }>
+    > => {
+        const imports: Array<{
+            target: string;
+            target_type: string;
+            source: string;
+            comment: string;
+            scope: Scope;
+        }> = [];
```

Comment:
Have you considered working stream-based here? This would prevent creating a large imports array. @nsams do you think this might ever be a problem in terms of memory usage?

Reply:
depends on the size of the import file. But I doubt we will ever have such large files

Reply:
Okay, then we'll ignore that for now.